### ISSUE TITLE
Role Naming Convention Mismatch leading to ye ole Not Authorized to access iam graphql from lambda function

### ISSUE THREAD

**Issue by joekiller** [aws-amplify/amplify-category-api#749]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.16.0

### Amplify CLI Version

9.2.1

### What operating system are you using?

Arch Linux 5.19.2-arch1-1

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

function, api

### Amplify Commands

update

### Describe the bug

The role name created by [autogeneratedParameters], `${appName}LambdaRole${shortId}`, doesn't match the convention expected by [getAdminRoles], `${r.resourceName}-${currentEnv}` in amplify-category-api leading to the set of `adminRoles` in the `Query.getModel.auth.1.req.vtl` being incorrect.

cfn in lambda
```
 "LambdaExecutionRole": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "RoleName": {
          "Fn::If": [
            "ShouldNotCreateEnvResources",
            "myappLambdaRolee80f9b07",
            {
              "Fn::Join": [
                "",
                [
                  "myappLambdaRolee80f9b07",
                  "-",
                  {
                    "Ref": "env"
                  }
                ]
              ]
            }
          ]
        },
```

vs the vtl:
```
#if( $util.authType() == "IAM Authorization" )
  #set( $adminRoles = ["myfunction-testenv"] )
```

This is with the new `"transformerversion": 2,` and `useexperimentalpipelinedtransformer": true`

[autogeneratedParameters]: https://github.com/aws-amplify/amplify-cli/blob/1a62c65eb086f6f5ec868419ef5758a637969f83/packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/autogeneratedParameters.ts#L16

[getAdminRoles]: https://github.com/aws-amplify/amplify-category-api/blob/a37e72e0c9d15eb7a75b0d2b6a56dfd06153c10b/packages/amplify-category-api/src/graphql-transformer/utils.ts#L57

### Expected behavior

I expect the two libraries to match conventions on role names.

### Reproduction steps

I guess do the steps in https://github.com/aws-amplify/amplify-category-api/issues/679

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @joekiller :wave: thanks for raising this! From the generated VTL resolvers we can see, as you noted, the `adminRoles` are constructed using the function's name, which is later verified by checking against the `userArn` of the caller's identity. When calling from a Lambda this ARN will include the `${functionName}-${envName}` naming convention.

Are you receiving an unauthorized access from the Lambda or from the GraphQL API? The linked issue describes an issue where the grant our Lambda function access to the GraphQL API but the resolvers are not regenerated to reflect the new set of `adminRoles` (containing our newly created function)

**Comment by joekiller** [comment]
@josefaidt thanks for the reply. I think I was getting unauthorized access when calling from the lambda function to a GraphQL endpoint. I'll double check this behavior and respond.

**Comment by joekiller** [comment]
@josefaidt I renamed my lambda role and it seems to be working regardless per the recognition of the lambda name arn, not the role name itself. Thanks for clarifying the behavior. I'm guessing I ran into perhaps not having applied the permissions to the lambda via the cli. I forgot that, as documented in [Working with Lambda execution environment credentials](https://docs.aws.amazon.com/lambda/latest/dg/lambda-intro-execution-role.html#permissions-executionrole-source-function-arn), that the lamda name is the "userName" provided to the authorizer so that makes sense.

I have a follow-up question though. If a lambda function within my environment is named via convention and given appropriate IAM permissions, is it sufficient to just include the Lambda name in "adminRoleNames" instead of the role name? The logic [in the template](https://github.com/aws-amplify/amplify-category-api/blob/2c2718284b557dcebb1c8de9f6021722be9ac60b/packages/amplify-graphql-auth-transformer/src/__tests__/__snapshots__/multi-auth.test.ts.snap#L10-L18) seems to indicate that should be enough and I shouldn't have to worry about the name of the role unless it is fully outside of my account.

My confusion of chasing the role name stemmed from the documentation covering lambdas generated by the cli or functions not deployed by the cli. [Grant Lambda function access to GraphQL API](https://docs.amplify.aws/cli/graphql/authorization-rules/#grant-lambda-function-access-to-graphql-api) under "the Function deployed without Amplify CLI" indicates that functions outside the cli must have their role included in the "adminRoleNames" custom-roles.json however that seems inaccurate. It seems that "a function deployed without amplify cli must have it's _name_ included in the custom-roles.json, not the role. Is this accurate?
